cypress/e2e/tests/administration/custom-migration-targets/trigger_validation.ts

```diff
@@ -0,0 +1,122 @@
+import { CustomMigrationTargetView } from "../../../views/custom-migration-target.view";
+import { CredentialsSourceControlUsername } from "../../../models/administration/credentials/credentialsSourceControlUsername";
+import { getRulesData } from "../../../../utils/data_utils";
+import { Analysis } from "../../../models/migration/applicationinventory/analysis";
+import { cancelButton } from "../../../views/common.view";
+import * as commonView from "../../../views/common.view";
+import { RulesRepositoryFields } from "../../../types/types";
+import { Application } from "../../../models/migration/applicationinventory/application";
+
+// Automates Bug MTA-3330 | Polarion TC MTA-597
+describe(["@tier3"], "Custom Migration Targets rules trigger validation", () => {
+    let target: CustomMigrationTarget;
+    const applications: Analysis[] = [];
+    const EXPECTED_EFFORT = 5;
+
+    before("Login", function () {
+        login();
+        deleteAllMigrationWaves();
+        deleteApplicationTableRows();
+    });
+
+    beforeEach("Fixtures and Interceptors", function () {
+        cy.fixture("application").then(function (appData) {
+            this.appData = appData;
+        });
+
+        cy.fixture("analysis").then(function (analysisData) {
+            this.analysisData = analysisData;
+        });
+
+        cy.fixture("custom-rules").then(function (customMigrationTargets) {
+            this.customMigrationTargets = customMigrationTargets;
+        });
+
+        CustomMigrationTarget.open(true);
+    });
+
+    it("Test same rules are triggered for custom rules and custom migration target", function () {
+        const targetData = this.customMigrationTargets["rules_from_bug_3330"];
+        target = new CustomMigrationTarget(
+            data.getRandomWord(8),
+            data.getDescription(),
+            targetData.image,
+            getRulesData(targetData),
+            Languages.Java
+        );
+        target.create();
+
+        for (let i = 0; i < 2; i++) {
+            const application = new Analysis(
+                getRandomApplicationData("Tackle Public", {
+                    sourceData: this.appData["tackle-testapp-public"],
+                }),
+                {
+                    source: "Source code",
+                    target: [],
+                }
+            );
+            application.create();
+            applications.push(application);
+        }
+
+        applications[0].target = [target.name];
+        applications[1].customRuleRepository = getRulesData(targetData) as RulesRepositoryFields;
+
+        applications[0].analyze();
+        applications[0].selectApplication();
+        applications[1].analyze();
+
+        applications.forEach((app) => {
+            app.verifyAnalysisStatus(AnalysisStatuses.completed);
+            Application.open(true);
+            app.verifyEffort(EXPECTED_EFFORT);
```

Comment:
@abrugaro I also want to validate that the custom rule is triggered not just the effort .
We can use the same app from the bug MTA-3330 if needed , https://github.com/selrahal/mta-bug and check if custom rule is triggered.

Reply:
@sshveta Here I'm running the analysis using the custom rules from the bug (https://github.com/selrahal/mta-bug) I just uploaded the same rules to the bookserver repo to have everything together (https://github.com/ibraginsky/book-server/tree/main/bug-3330-rules).

Since I'm analyzing the app with just these rules, validating the effort is equivalent to validate the triggering of the rules because I analyze both apps with the same rules that should be equally triggered, hence, report the same effort.